FHEM/14_SD_BELL.pm

```diff
@@ -166,9 +176,26 @@ sub Set($$$@) {
 
 	if ($cmd eq "?") {
 		$ret .= "ring:noArg";
+		$ret .= " Alarm:noArg" if ($protocol == 96);	# only Grothe_Mistral_SE
 	} else {
-		my $rawDatasend = $split[1];													# hex value from def without protocol
-		if ($rawDatasend =~ /[0-9a-fA-F]_[0-9a-fA-F]/s) {			# check doubleCode in def
+		if ($protocol == 96) {;	# only Grothe_Mistral_SE
+			# set sduino434 raw SC;;R=5;;SR;;R=1;;P0=1500;;P1=-215;;D=01;;SM;;R=1;;C=215;;D=47104762003F;;
+			my $msg = "SC;;R=";
+			$msg .= $repeats;
+			$msg .= ";;SR;;R=1;;P0=1500;;P1=-215;;D=01;;SM;;R=1;;C=215;;D=47";
+			my $id = $split[1];
+			$id = sprintf('%06X', hex(substr($id,0,6)) | 0x800000) if ($cmd eq "Alarm");	# set alarm bit
+			$msg .= $id;
+			my $checksum = sprintf('%02X', ((0x47 + hex(substr($id,0,2)) + hex(substr($id,2,2)) + hex(substr($id,4,2))) & 0xFF));
+			$msg .= $checksum;
+			my $model = AttrVal($name,'model', 'Grothe_Mistral_SE_01');
+			$msg .= "3F" if ($model eq "Grothe_Mistral_SE_03");	# only Grothe_Mistral_SE_03
+			$msg .= ";;";
+			fhem("set $ioname raw $msg");
```

Comment:
Wieso rufst Du einen Set Befehl auf? Ich verstehe aktuell nicht so ganz, was hier passiert.
:(

Reply:
Moin, ich rechnete mit der Frage.

Dieser Set Befehl wird aufgerufen weil wenn ich es mit der Funktion direkt machen würde (add..., wie diese heißt weiß ich zum frühen Morgen nicht 100%) so muss ich den Hash vom Empfänger übergeben. Da dies mit mehreren Zeilen erst wieder herausgesucht werden muss über IODev, so wurde der kurze Weg genommen über den setbefehl.

Nicht :(, ich ergänze auch gern auf den Hash damit die Zeile ersetzt wird. :)

Reply:
@sidey79 ,
nachdem ich nochmal am Tag drüber nachgedacht habe, so fiel mir wieder ein wieso ich es so umgesetzt hatte. Uns gelang absolut nicht die Übergabe zum IODev. Wie würdest du es lösen oder haben wir da vielleicht sogar eine Nichtfunktion gefunden.

Bisheriger Versuch, die direkte Übergabe an die add... Routine in der SetFunktion vom Signalduino.

Reply:
Verstehe.

Für die Kommunikation zwischen logischem Modul und Physischem Modul sieht FHEM eine Schnittstelle vor.

https://wiki.fhem.de/wiki/DevelopmentModuleIntro#X_Write

Die Funktion wird durch das SIGNALDuino Modul wie folgt definiert:
https://github.com/RFD-FHEM/RFFHEM/blob/fd0bd1e515fb9fadb20dcf9f5461cf9913fe7fc0/FHEM/00_SIGNALduino.pm#L1287-L1311

Reply:
Ich denke mal, das diese Änderung nun deinen Vorstellungen entspricht und so richtig umgesetzt wurde.